app/googleanalytics/pixels.py

```diff
@@ -1,35 +1,53 @@
-from urllib.parse import urlencode
-
+import os
 from flask import current_app
+from urllib.parse import quote
+
+TEST_DOMAIN = 'https://test-api.va.gov/vanotify/'
+DEV_DOMAIN = 'https://dev-api.va.gov/vanotify/'
+PERF_DOMAIN = 'https://sandbox-api.va.gov/vanotify/'
+STAGING_DOMAIN = 'https://staging-api.va.gov/vanotify/'
+PROD_DOMAIN = 'https://api.va.gov/vanotify/'
+
+NOTIFICATION_API_GA4_GET_ENDPOINT = 'ga4/open-email-tracking'
+
+GA4_PIXEL_TRACKING_NAME = 'email_open'
+GA4_PIXEL_TRACKING_SOURCE = 'vanotify'
+GA4_PIXEL_TRACKING_MEDIUM = 'email'
+
 
+def get_domain_for_environment():
+    environment = os.getenv('NOTIFY_ENVIRONMENT', 'development')
 
-def build_ga_pixel_url(
-    notification,
-    provider,
-):
-    url_params_dict = {
-        'v': '1',
-        't': 'event',
-        'tid': current_app.config['GOOGLE_ANALYTICS_TID'],
-        'cid': notification.id,
-        'aip': '1',
-        'ec': 'email',
-        'ea': 'open',
-        'el': notification.template.name,
-        'dp': f"/email/vanotify"
-        f"{'/' + notification.service.organisation.name if notification.service.organisation else ''}"
-        f"/{notification.service.name}"
-        f"/{notification.template.name}",
-        'dt': notification.subject,
-        'cn': notification.template.name,
-        'cs': provider.get_name(),
-        'cm': 'email',
-        'ci': notification.template.id,
+    ENVIRONMENT_DOMAINS = {
+        'test': TEST_DOMAIN,
+        'development': DEV_DOMAIN,
+        'perf': PERF_DOMAIN,
+        'staging': STAGING_DOMAIN,
+        'prod': PROD_DOMAIN,
     }
 
-    url_str = current_app.config['GOOGLE_ANALYTICS_URL']
-    url_params = urlencode(url_params_dict)
-    url = f'{url_str}?{url_params}'
+    return ENVIRONMENT_DOMAINS[environment]
+
+
+def build_dynamic_ga4_pixel_tracking_url(notification):
+    """
+    Constructs a dynamic URL that contains information on the notification email being sent.
+    The dynamic URL is used for pixel tracking and sends a request to our application when
+    email is opened.
+
+    :param notification: The notification object containing template and service details.
+    :return: A dynamically constructed URL string.
+    """
+
+    domain = get_domain_for_environment()
+    ga4_url = NOTIFICATION_API_GA4_GET_ENDPOINT
+    url = (
+        f'{domain}'
+        f'{ga4_url}?'
+        f'campaign={quote(notification.template.name)}&campaign_id={quote(str(notification.template.id))}&'
+        f'name={quote(GA4_PIXEL_TRACKING_NAME)}&source={quote(GA4_PIXEL_TRACKING_SOURCE)}&medium={quote(GA4_PIXEL_TRACKING_MEDIUM)}&'
+        f'content={quote(notification.service.name)}/{quote(str(notification.service.id))}/{quote(str(notification.id))}'
+    )
 
-    current_app.logger.info(f'Generated google analytics pixel URL: {url}')
+    current_app.logger.info(f'Generated Google Analytics 4 pixel URL: {url}')
```

Comment:
No f-strings for logging, please. see [style guide](https://github.com/department-of-veterans-affairs/vanotify-team/blob/master/Process/style_guide.md).

Reply:
Updated.